src/SetTypeTrait.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Phlib\Mail;
+
+/**
+ * Trait to enable the content-type to be overridden
+ *
+ * @package Phlib\Mail
+ */
+trait SetTypeTrait
```

Comment:
I've the behaviour to use immutable MIME types in #4 so this branch isn't responsible for changing the construct params of un-related classes. Then I can update the implementation here.